src/poetry/utils/authenticator.py

```diff
@@ -134,9 +136,11 @@ def __init__(
             self._get_repository_config_for_url
         )
         self._pool_size = pool_size
+        self.user_agent = user_agent("poetry", __version__)
```

Comment:
I'd either make it a property (as in `Uploader`) or prepend an underscore (`self._user_agent`).